iced-hs.cabal

```diff
@@ -0,0 +1,421 @@
+cabal-version: 3.8
+name: iced-hs
+version: 0.0.3.0
+homepage: https://github.com/ibaryshnikov/iced-hs
+bug-reports:
+  https://github.com/ibaryshnikov/iced-hs/issues
+
+license: MIT
+build-type: Custom
+tested-with:
+  ghc ==9.6.6 || ==9.8.4 || ==9.10.1
+
+extra-source-files:
+  LICENSE
+  README.md
+
+flag examples
+  description: Build examples
+  manual: True
+  default: False
+
+source-repository head
+  type: git
+  location: https://github.com/ibaryshnikov/iced-hs
+
+custom-setup
+  setup-depends:
+    base        >= 4.18 && < 5,
+    Cabal       >= 3.8 && < 3.15,
+    directory,
+    process,
+
+common common
+  build-depends:
+    base >=4.14 && <5,
+    bytestring,
+    template-haskell,
+
+  default-extensions:
+    CApiFFI
```

Comment:
`CApiFFI` is not needed for now since there's only `ccall` in the code. Check other comments for more details